docs/how-to-guides/create-a-user-defined-function.md

```diff
@@ -400,6 +400,14 @@ There are many things to observe in this class:
   you're using session windows, consider what good merge semantics are
   for your aggregation.
 
+#### Dynamic UDAFs
+
+If a UDAF's aggregate or return types vary based on the input type, one can either write a separate
+function annotated with @UdafFactory per type or override the following three methods
+`initializeTypeArguments(List<SqlArgument> argTypeList)`, `getAggregateSqlType()`, and
+`getReturnSqlType()`.  To see a concrete example in the ksqlDB codebase, check out the 
+implemenation of `latest_by_offset` or `collect_list`.
```

Comment:
It would great to add links:
- https://github.com/confluentinc/ksql/blob/master/ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/offset/LatestByOffset.java
- https://github.com/confluentinc/ksql/blob/master/ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/array/CollectListUdaf.java